### ISSUE TITLE
Shaders requiring Quadmesh do not work properly

### ISSUE THREAD

**Issue by stryker313** [godotengine/godot#55275]
### Godot version

v4.0.dev.20211117.official

### System information

Windows 10, gtx 1060, i5 4690k

### Issue description

Shaders that require a 2x2 quadmesh with max cull do not funciton properly in Godot 4 prealpha builds

Here is a photo of a shader in Godot 3. It is a simple post process outline
![Desktop Screenshot 2021 11 23 - 18 57 31 06](https://user-images.githubusercontent.com/83078010/143148018-d837981f-97a8-49fd-94f0-139fc0418fa1.png)

Here is a photo in Godot 4.
![Desktop Screenshot 2021 11 23 - 19 02 27 74](https://user-images.githubusercontent.com/83078010/143148024-20cfe5f4-b3ac-4698-9d36-6fcea51d1685.png)

There are no apparent errors in the shader, but running in the game and editor there is no effect.

I have tested various shaders, including
PS1 shader - https://godotshaders.com/shader/ps1-post-processing/
Dither shader
Other post process outline shaders - https://github.com/Ainlid/godot3.1.1depthedge

This seems to affect all shaders requiring a 2x2 quadmesh.(Post process shaders)
There is simply no resulting effect.

I do believe this is a Godot side glitch, beacuse I was able to achieve this
![Desktop Screenshot 2021 11 23 - 17 20 24 12](https://user-images.githubusercontent.com/83078010/143148127-97faca06-b950-4a9e-b27b-cbe5093c7a5d.png)
with messing around with render_mode, although it may just be a random side effect

### Steps to reproduce

1. Create a new game
2. load a cube
3. Load a camera
4. Load a quadmesh
5. Set the quad mesh to size 2x2
6. Load whatever shader you like onto the quadmesh
7. Results

### Minimal reproduction project

[GodotShaderBug.zip](https://github.com/godotengine/godot/files/7591689/GodotShaderBug.zip)

**Comment by clayjohn** [comment]
Did you update the shader at all? Note especially that the range of depth changed in 4.0 as described here https://docs.godotengine.org/en/latest/tutorials/shaders/advanced_postprocessing.html#depth-texture

**Comment by stryker313** [comment]
Not really. Although, after more testing, I have noticed that render_mode cull_disabled does help fix issues in regards to certain shaders (haven't tested on all of them, only PP outlines)

**Comment by moorejs** [comment]
I reproduced this issue when using a similar outline shader. I narrowed it down to one line, the `VIEWPORT_SIZE`, not being set correctly ([code](https://github.com/jocamar/Godot-Post-Process-Outlines/blob/12a9e0e9b2e31ba71c9feb0734e23fab50a00b98/addons/jm_pp_outlines/post_process_outlines.shader#L36)). I set it to e.g. 1920 and 1080 and the shader works as expected.

The PS1 shader uses `FRAGCOORD` so I assume there may be a similar issue there.

It seems both these issues are discussed in #51546 and #47711.

As far as I can tell, my `VIEWPORT_SIZE` is nearly `(0.08, 0.04)` for 1920x1080.

**Comment by Zylann** [comment]
My planet atmosphere shader is broken as well in Godot4.

It has two modes: box rendering when far enough, and full-screen quad when close enough (it dynamically switches to remain visible and optimize cost, especially when multiple atospheres are visible).
The latter full-screen quad mode does not render anything https://github.com/Zylann/godot_atmosphere_shader/blob/b0e27e6e6b1943545151110eeb312079e295c58b/addons/zylann.atmosphere/planet_atmosphere.shader#L83

**Comment by Zireael07** [comment]
Just ran into this issue on a **canvas item** (tried to make a hud overlay). Tracked it down to me using FRAGCOORD (outputting COLOR = FRAGCOORD.xy, 0.0 gives me totally uniform yellow)

Something is def fishy there, please fix ASAP as FRAGCOORD is very commonly used, especially if people port shaders from Shadertoy (my hud overlay thingy is based on a Shadertoy port)

**Comment by clayjohn** [comment]
@Zireael07 I think your issue may be unrelated. ``COLOR.xy = FRAGCOORD.xy`` will give uniform yellow except for a one pixel border because FRAGCOORD is in viewport space e.g. it is in pixels, so it will almost always have a value greater than 1

**Comment by Zireael07** [comment]
The same FRAGCOORD in 3.x works as expected (.xy outputs green to red colors, similar to UV)

As I said, FRAGCOORD is commonly used in Shadertoy shaders instead of UV. And the shading guide still recommends FRAGCOORD for porting shadertoy stuff
